### ISSUE TITLE
Points to add

### ISSUE THREAD

**Issue by sara-02** [nickedes/Pybaby#2]
1. ~~Zip and argument packing~~
2. ~~Conditions~~
3. ~~Random value generation~~
4. Concept of generators and lazy range
5. ~~kwargs~~
6. Regex
7. Classes and main method.
8. Try-catch; exception handling.

**Comment by nickedes** [comment]
We can add these too : 
9. Reading nd writing json files.
10. Converting one Data-structure to another (For example : creating a set of value in dictionary etc.)'

**Comment by antiDigest** [comment]
Also add some numpy methods like:
array()
ndarray()
matrix()
add/subtract/divide/multiply etc.

**Comment by zishan0215** [comment]
Do you expect them to study all this by themselves?
I think it's better if we have a session on this.. Nikhil can be our guest lecturer.

On 5 Mar 2016 8:34 a.m., Antriksh Agarwal notifications@github.com wrote:

Also add some numpy methods like:
array()
ndarray()
matrix()
add/subtract/divide/multiply etc.

**Comment by antiDigest** [comment]
No, session on this is useless, they only need to know a few functions/methods most used, might tell them that directly, but no use of wasting a session on this.

**Comment by zishan0215** [comment]
Then take a test. Because no one will read this on their own.

On 5 Mar 2016 9:29 a.m., Antriksh Agarwal notifications@github.com wrote:

No, session on this is useless, they only need to know a few functions/methods most used, might tell them that directly, but no use of wasting a session on this.

**Comment by sara-02** [comment]
We can have test for the basic Python, covering the points in the readme plus the ones listed by me and @nickedes. For numpy we can have a session during break. We cannot expect them to cover up so much on their own.

**Comment by antiDigest** [comment]
Or maybe just mark on the assignment, because the assignment uses both python and numpy and pandas as well.
